qiskit/providers/aer/backends/cluster/clusterjobset.py

```diff
@@ -0,0 +1,236 @@
+
+    def run(self,
+            executor: futures.Executor,
+            backend: AerBackend,
+            *run_args: Any,
+            **run_kwargs: Any) -> None:
+        """Execute this set of jobs on an executor.
+
+        Args:
+            executor: The executor used to submit jobs asynchronously.
+            backend: The backend to run the jobs on.
+            *run_args: Positional arguments passed through to backend.run.
+            **run_kwargs: Keyword arguments passed through to backend.run.
+
+        Raises:
+            RuntimeError: If the jobs were already submitted.
+        """
+        if self._futures:
+            raise RuntimeError(
+                'The jobs for this managed job set have already been submitted.')
+
+        self._future = True
+        total_jobs = len(self._experiments)
+        for i, exp in enumerate(self._experiments):
+            cjob = CJob(backend, exp, *run_args, **run_kwargs)
+            cjob.submit(executor=executor)
+            logger.debug("Job %s submitted", i + 1)
+            self._futures.append(cjob)
+
+    @requires_submit
+    def statuses(self) -> List[Union[JobStatus]]:
+        """Return the status of each job in this set.
+
+        Returns:
+            A list of job statuses.
+        """
+        return [cjob.status() for cjob in self._futures]
+
+    @requires_submit
+    def result(
+            self,
+            timeout: Optional[float] = None,
+            raises: Optional[bool] = False
+    ) -> Result:
+        """Return the results of the jobs as a single Result object.
+
+        This call will block until all job results become available or
+        the timeout is reached.
+
+        Args:
+           timeout: Number of seconds to wait for job results.
+           raises: whether or not to re-raise exceptions thrown by jobs
+
+        Returns:
+            qiskit.Result: Result object
+
+        Raises:
+            AerClusterTimeoutError: if unable to retrieve all job results before the
+                specified timeout.
+
+        """
+        return self.results(timeout, raises).combine_results()
+
+    @requires_submit
+    def results(
+            self,
+            timeout: Optional[float] = None,
+            raises: Optional[bool] = False
+    ) -> CResults:
+        """Return the results of the jobs.
+
+        This call will block until all job results become available or
+        the timeout is reached. Analogous to dask.client.gather()
+
+        Args:
+           timeout: Number of seconds to wait for job results.
+           raises: whether or not to re-raise exceptions thrown by jobs
+
+        Returns:
+            A :class:`CResults`
+            instance that can be used to retrieve results
+            for individual experiments.
+
+        Raises:
+            AerClusterTimeoutError: if unable to retrieve all job results before the
+                specified timeout.
+        """
+        if self._results is not None:
```

Comment:
I think this method just needs to call the individual worker `job.result()` something like:

```python
if isinstance(worker, Integral):
    return self._futures[worker].result(timeout=timeout, raises=raises)
else:
    return [self._futures[i].result(timeout=timeout, raises=raises) for i in worker]
```